traffic_ops/traffic_ops_golang/api/shared_handlers.go

```diff
@@ -298,23 +298,36 @@ func DeleteHandler(deleter Deleter) http.HandlerFunc {
 		obj := reflect.New(objectType).Interface().(Deleter)
 		obj.SetInfo(inf)
 
-		keyFields := obj.GetKeyFieldsInfo() // expecting a slice of the key fields info which is a struct with the field name and a function to convert a string into a interface{} of the right type. in most that will be [{Field:"id",Func: func(s string)(interface{},error){return strconv.Atoi(s)}}]
-		keys := make(map[string]interface{})
-		for _, kf := range keyFields {
-			paramKey := inf.Params[kf.Field]
-			if paramKey == "" {
-				HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("missing key: "+kf.Field), nil)
-				return
+		deleteKeyOptionExists := false
+		if d, ok := obj.(HasDeleteKeyOptions); ok {
+			options := d.DeleteKeyOptions()
+			for key, _ := range options {
+				if inf.Params[key] != "" {
+					deleteKeyOptionExists = true
+					break
+				}
 			}
```

Comment:
should this error as they said it has delete key options but none was specified in the params?

Reply:
Having that error makes sense to me.

Reply:
Made specifying a delete key in 2ada94e8f2.